sysl2/sysl/visitor.go

```diff
@@ -52,15 +53,15 @@ func MakeEndpointCollectionElement(title string, endpoints []string, blackboxes
 		uptos = append(uptos, fmt.Sprintf("%s <- %s", entry.appName, entry.endpointName))
 	}
 
-	bb := make(map[string]string)
-	for _, b := range blackboxes {
-		switch len(b) {
+	bb := make(map[string]*Upto)
+	for k, b := range blackboxes {
+		switch len(b.Comment) {
 		case 0:
-			continue
 		case 1:
-			bb[b[0]] = ""
+			b.Comment = ""
+			bb[k] = b
 		default:
-			bb[b[0]] = b[1]
+			bb[k] = b
```

Comment:
On second look, maybe this should be:
```go
		if b.Comment > "" {
			bb[k] = b
		}
		if b.Comment == "#" {
			b.Comment = ""
		}
```

Reply:
Changed but the 2nd if is not needed in my opinion

Reply:
Changed the case logic to simplify it.